src/components/DailyConfigComponent.tsx

```diff
@@ -51,91 +72,171 @@ const DailyConfig = () => {
+                      if (enabled) {
+                        const deviceId = currentDevice || nanoid();
+                        setConfig({ device: deviceId });
+                        localStorageSet("device", deviceId);
+                        setCurrentDevice(deviceId);
+                      } else {
+                        setConfig({ device: "" });
+                      }
+                    }}
+                    disabled={!enabled}
+                  />
+                </FormGroup>
+              </div>
+            }
+            disabled={!enabled}
+          />
+        )}
+        {enabled && (
+          <Tab
+            className="text-white"
+            id="rdt"
+            title="Run Details"
+            panel={
+              <div className="flex items-start gap-2 flex-col">
+                <Label>
+                  Last Run
+                  <Text style={{ color: "#8A9BA8" }}>
+                    {lastRun && lastRun !== "01-01-1970"
+                      ? `Last ran on page ${lastRun}`
+                      : "Last run data not available"}
+                  </Text>
+                </Label>
+                <Label>
+                  Next Run
+                  <Text style={{ color: "#8A9BA8" }}>
+                    {!!nextRun
+                      ? `${Math.floor(
+                          nextRun / (60 * 60 * 1000)
+                        )} hours, ${Math.floor(
+                          (nextRun % (60 * 60 * 1000)) / (60 * 1000)
+                        )} minutes, ${Math.floor(
+                          (nextRun % (60 * 1000)) / 1000
+                        )} seconds`
+                      : "Next run data not available"}
+                  </Text>
+                </Label>
+                <Label>
+                  Current Date and Time
+                  <Text style={{ color: "#8A9BA8" }}>
+                    {new Date().toLocaleString()}
+                  </Text>
+                </Label>
+                <Label>
+                  Next Run Scheduled At
+                  <Text style={{ color: "#8A9BA8", maxWidth: "200px" }}>
+                    {new Date(configNextRun).toLocaleString()}
+                  </Text>
+                </Label>
+                <Label>
+                  Set "Time To Run"
+                  <Text style={{ color: "#8A9BA8", maxWidth: "200px" }}>
+                    {timeToRun
+                      .split(":")
+                      .map((val, i) => (i === 0 ? `${val} HR` : `${val} M`))
+                      .join(" ")}
+                  </Text>
+                </Label>
+              </div>
+            }
+          />
+        )}
+        <Tabs.Expander />
+        <Switch
+          large={true}
+          className="w-full text-right"
+          checked={enabled}
+          onChange={async (e) => {
+            const enabled = (e.target as HTMLInputElement).checked;
+            if (enabled) {
+              await setConfig({
```

Comment:
this `await` might mitigate some users problems

https://www.loom.com/share/bca4d70c4f1546478659b81667a5e55c